### DISCUSSION TITLE
API Rules

### DISCUSSION THREAD

**Discussion by nexus9111** [pocketbase/pocketbase#5273] [Q&A] [answered]
## Users Collection

I expanded the basic users collection with 2 fields:
- role: select[]
- public: boolean

## Companies Collecion

- id: auto
- name: string
- users: users[]

## My difficulty

I would like to edit the users view rule to make a logged in user able to get information about himself, public users, all users in his company. I don't succeed the last one, here is my actual user view rule (the users in a company are not public):

```
@request.auth.id != "" && (
    id = @request.auth.id || 
    public = true || 
    (
      companies_via_users.users.id ?= @request.auth.id && 
      companies_via_users.users.id ?= id
    )
)
```

When request companies and expand users, it only retrieve himself

Thanks a lot

**Comment by ganigeorgiev** [comment, marked as answer]
The 2 `companies_via_users.users.id` expressions target a single id value and it cannot be 2 different things (`@request.auth.id` and `id`) at the same time (the reason is similar to https://github.com/pocketbase/pocketbase/discussions/4319).

But if I'm understanding your collections structure I don't think you need to do that. Your API rule could look like:

```js
@request.auth.id != "" && (
    id = @request.auth.id || 
    public = true || 
    @request.auth.companies_via_users.users.id ?= id
)
```

**Reply by nexus9111** [reply]
Ohhh thanks ! I didn't understand this well ! Thanks a lot
